.github/workflows/cypress-test-oidc-e2e.yml

```diff
@@ -0,0 +1,192 @@
+name: Snapshot based E2E OIDC tests workflow
+on:
+  pull_request:
+    branches: [ '**' ]
+env:
+  OPENSEARCH_VERSION: '3.0.0'
+  KEYCLOAK_VERSION: '21.0.1'
+  TEST_KEYCLOAK_CLIENT_SECRET: 'oacHfNaXyy81r2uHq1A9RY4ASryre4rZ'
+  CI: 1
+  # avoid warnings like "tput: No value for $TERM and no -T specified"
+  TERM: xterm
+  PLUGIN_NAME: opensearch-security
+  
+jobs:
+  tests:
+    name: Run Cypress E2E OIDC tests
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest ]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      
+      - name: Checkout Branch
+        uses: actions/checkout@v3
+
+      - name: Set env
+        run: |
+          opensearch_version=$(node -p "require('./package.json').opensearchDashboards.version")
+          plugin_version=$(node -p "require('./package.json').version")
+          echo "OPENSEARCH_VERSION=$opensearch_version" >> $GITHUB_ENV
+          echo "PLUGIN_VERSION=$plugin_version" >> $GITHUB_ENV
+        shell: bash
+
+      # Setup and Run Keycloak
+      - name: Get and run Keycloak on Linux
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          echo "Downloading Keycloak ${{ env.KEYCLOAK_VERSION }}"
+          wget https://github.com/keycloak/keycloak/releases/download/${{ env.KEYCLOAK_VERSION }}/keycloak-${{ env.KEYCLOAK_VERSION }}.tar.gz
+          echo "Unpacking Keycloak"
+          tar -xzf keycloak-${{ env.KEYCLOAK_VERSION }}.tar.gz
+          export KEYCLOAK_ADMIN=admin
+          export KEYCLOAK_ADMIN_PASSWORD=admin
+          cd keycloak-${{ env.KEYCLOAK_VERSION }}/bin
+          chmod +x kc.sh
+          echo "Starting keycloak"
+          ./kc.sh start-dev --http-enabled=true --hostname-strict-https=false --http-host=localhost --http-relative-path /auth --health-enabled=true &
```

Comment:
This is unsafe - download a file and make it executable during in GHA exposes our repository to all kinds of bad behavior if the source is compromised.

Can we validate the download with a checksum to be sure only the expected version is allowed?

Reply:
The version check before actual run has been added.

Reply:
Consider if if some how the zip file was edited by a malicious user they could add any code such a crypto mining or other malicious software that would execute.  The version would come back the same, but the contents of the zip aren't known to us.

By having a checksum in this file, we would have to verify it when doing a version bump.

Reply:
Ah I fully understand now. I have generated a SHA256 checksum, and compare it with the download one before making the script become executable.

If the they are not matching, then the output should be something like this: https://github.com/opensearch-project/security-dashboards-plugin/actions/runs/7036456741/job/19149098718?pr=1579#step:5:3344